src/field_ext.rs

```diff
@@ -30,10 +30,18 @@ impl FieldExt for Fq {
         debug_assert_eq!(self.serialized_size(), 32);
         self.serialize(&mut bytes[..])
             .expect("serialization into array should be infallible");
+        // Set top three bits of last byte to zero
+        bytes[31] &= 0b00011111;
+
         bytes
     }
 
     fn from_bytes(bytes: [u8; 32]) -> Result<Self, EncodingError> {
+        // Top three bits of last byte should be zero
```

Comment:
It'd be helpful to note in this comment that we only do this check because Arkworks *doesn't* consider the top three bits when deciding whether or not an encoding is canonical.